pkg/cri/store/sandbox/status.go

```diff
@@ -45,9 +45,9 @@ import (
 // |                  |              |           |
 // |                  | Stop/Exit    |           |
 // |                  |              |           |
-// |             +----v----+         |           |
-// |             |         <---------+      +----v----+
-// |             | NOTREADY|                |         |
+// |   cleanup   +----v----+         |           |
+// |   (failed)  |         <---------+      +----v----+
+// |------------>| NOTREADY|                |         |
```

Comment:
seems there are two not ready states.. not ready with ready to be deleted, and and not ready with needs more cleanup (another stop attempt)

Reply:
The comments for StateNotReady below should be updated as well.

https://github.com/containerd/containerd/blob/f242de03d1bfc77ec5962c8a9dcf39ceafe69cb5/pkg/cri/store/sandbox/status.go#L70-L75

Reply:
> seems there are two not ready states.. not ready with ready to be deleted, and not ready with needs more cleanup (another stop attempt)

Yeah.

In kubelet, we have to make sure that each sandbox remove operation has a successfully sandbox stop before it.

Reply:
From my code search, it works like way. https://github.com/kubernetes/kubernetes/blob/v1.24.1/pkg/kubelet/kuberuntime/kuberuntime_gc.go#L179